infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -608,222 +748,186 @@ public void setReplicationStatusTable(String clusterId, long remainingEntries, S
                         .build();
             }
 
-            current = ReplicationStatusVal.newBuilder()
-                    .setRemainingEntriesToSend(remainingEntries)
-                    .setSyncType(type)
-                    .setStatus(SyncStatus.ONGOING)
-                    .setSnapshotSyncInfo(currentSnapshotSyncInfo)
+            current = ReplicationStatus.newBuilder()
+                    .setSourceStatus(SourceReplicationStatus.newBuilder()
+                            .setRemainingEntriesToSend(remainingEntries)
+                            .setReplicationInfo(ReplicationInfo.newBuilder()
+                                    .setSyncType(type)
+                                    .setStatus(SyncStatus.ONGOING)
+                                    .setSnapshotSyncInfo(currentSnapshotSyncInfo)
+                                    .build())
+                            .build())
                     .build();
 
             try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
-                txn.putRecord(replicationStatusTable, key, current, null);
+                txn.putRecord(statusTable, session, current, null);
```

Comment:
Should this happen transactionally with the check above in line 692? Because if the session data has changed between 692 and 763 this update can potentially overwrite a conflicting update right?